microsoft/atlas-design
# Add color atomic classes

[task-378401](https://dev.azure.com/ceapex/Engineering/_boards/board/t/Fox%20Team/Stories/?workitem=378401)

This PR adds color atomic classes with the format: `.cssproperty-value-screensize`

With a mobile first approach, the screensize being used are tablet, desktop, and widescreen.
- Although I don't think the widescreen options will be used often, I added it for maximum flexibility.
- Brand colors can't be added until support for IE is dropped.

### Background colors
- `has-body-background` is now `background-color-main`.
- Added alternate background color, which is the opposite of the current theme. If you're currently on light theme, the alternate is black. 
- The light theme `background-color-main` step becomes darker from `main -> main-medium -> main-dark`.  The alternate background starts at black and becomes lighter `alternate -> alternate-medium -> alternate-light`. (main: white >>> center/dark grey <<< alternate: black). 
- For dark and high contrast theme, the main and alternate starts at the center instead and steps toward the two ends of the spectrum. (black <<< main: dark grey, alternate:light grey >>> white)
- Per the [Figma](https://www.figma.com/proto/03HGueCJEx4tyZd3Rhfw5B/DevRel-Design-System?node-id=268%3A343&viewport=1143%2C-507%2C0.08135364204645157&scaling=scale-down-width), there will be only two shades of dark grey for the background color (vs the three currently in use). 

## Testing

1. Open "helpers > colors.scss" in docs-ui and compare with the changes in this PR. 
2. To test it live, visit this [PR in docs-ui](https://dev.azure.com/ceapex/Engineering/_git/docs-ui/pullrequest/157695) . Follow the steps in its description.
3. Checkout `atlas-test-3-23` branch in docs-ui. Go to `packages/styles/src/utilities/_theme.scss`. Add the following to each theme map:
```
'light': 	'body-background-alt': $system-grey-100,
		'body-background-alt-medium': $system-grey-90,
		'body-background-alt-light': $system-grey-80,

'dark': 	'body-background-dark': $system-grey-100,   //temporarily replace the existing value to match the new value in Atlas for testing purposes
                'body-background-alt': $system-grey-05,
		'body-background-alt-medium': $system-grey-02,
		'body-background-alt-light': $system-grey-00,

'high contrast': 'body-background-alt': $system-grey-05,
		'body-background-alt-medium': $system-grey-02,
		'body-background-alt-light': $system-grey-00,
```

Note: these colors should be the equivalent to the ones being added in the Atlas theme map.
4. Run `yarn content-reset; yarn develop`. 
5. Visit http://localhost:3000/en-us/test/docs/conceptual/azure-cli or any other page. Scroll down to the Feedback section.
6. Open a similar prod page in another tab to compare, https://docs.microsoft.com/en-us/dotnet/core/dotnet-five . 
7. Notice how the feedback section background color in the prod page page uses `--theme-body-background-dark`, yet it's lighter than the surrounding colors (`*-background-medium`). Compare with the localhost page. `--theme-body-background-dark` is now actually darker (black). 
8. Experiment by inspecting an element in the localhost page with DevTools and add the new classes or replace existing classes with the new classes. Check other pages.


old feedback:
![image](https://user-images.githubusercontent.com/64044661/113221815-9d50aa80-923a-11eb-90d2-237ee0769ae5.png)

new feedback:
![image](https://user-images.githubusercontent.com/64044661/113221941-d557ed80-923a-11eb-825d-6629fc6ac0c3.png)
